### ISSUE TITLE
[Issue Feedback] Some confusion about event controller

### ISSUE THREAD

**Issue by JettLand** [Regalis11/Barotrauma#2395]
- [x] I have searched the issue tracker to check if the issue has already been reported.

**Description**
I'm making a difficulty enhance mod based on monster spawn.But for unknow reasons,it works not well.
I'v edited 3 xml files:'missions.xml' ,'randomevents.xml' and 'eventmangersettings.xml'.
Only [spawntype="ruin"] section is working,the [spawntype="mainpath"] section seems nothing happened in game.
I'v been try so many ways to figure out the issue,and check my codes for syntax errors.
Anyway,still same as before.
These above conclusions are summarized in multiplayer games.
I'll appreciate if anyone can solve my confusion,thank a lot!

**Steps To Reproduce**
Just read the codes.

**Version**
Barotrauma v0.9.7.0, Windows 7 Ultimate edition.

**Additional information**
[Missions.txt](https://github.com/Regalis11/Barotrauma/files/4292989/Missions.txt)
[Randomevents.txt](https://github.com/Regalis11/Barotrauma/files/4292991/Randomevents.txt)
[EventManagerSettings.txt](https://github.com/Regalis11/Barotrauma/files/4292993/EventManagerSettings.txt)

**Comment by JettLand** [comment]
> 嗨！我认为“最小强度”值是这里的问题。强度是介于0-1之间的值，表示当前情况有多严重（例如，当潜艇泛滥，附近有怪物或某些机组人员受伤时，强度会增加）。例如，如果将minintensity设置为0.5，则意味着该事件的激活强度必须至少为0.5（即，除非已经有很多动作发生，否则怪物不会生成）。maxintensity值的工作方式相反：如果值为0.5，则强度为0.5或更大时，偶数将不会激活（即，如果情况已经很糟，则生成会延迟）。
> 
> 在您的配置文件中，最小强度值很高，有些甚至大于1，这意味着它们将永远不会被激活。尝试将其设置为0（或完全删除最小强度参数），事件应开始工作。
> 
> 另外，我们在默认的随机事件配置中添加了一些注释，这可能会有所帮助（当前公共版本中尚未包括）：
> 
> ## 事件集
> **chooserandom** = true/false. If true, one of the events in the set is chosen randomly, otherwise all of them are used.
> **mindistancetraveled** = 0-1. How large portion of the level must the submarine traverse before the event can be activated.
> **minmissiontime** = How many seconds must have passed before the event can be activated. Note that the event activates if either mindistancetraveled OR minmissiontime is reached. If you want to make the event purely time-based or only distance-based, you can set the distance or time limit to a very high value.
> **minintensity, maxintensity** = 0-1. Eventmanager's intensity value must be within this range for the event to be activated. Use the minintensity value to prevent the event from occurring when intensity is low (i.e. when there's no immediate danger or much going on in the sub), and the maxintensity value to prevent activating additional events if there crew is already in trouble.
> 
> ## Any type of event
> **musictype** = a music tag that matches a value defined in the sound config (e.g. monster, deep, flooded, levelend, ruins...)
> 
> ## MonsterEvent
> **characterfile** = the identifier of the monster to spawn
> **amount** = how many of the monster to spawn
> **minamount, maxamount** = use to configure a random number of monsters to spawn
> **spawntype** = MainPath, Cave, Ruin. Where to spawn the monster(s).
> **spawndeep** = true/false. When set to true, the monster spawns below the level.

I apologize for my understanding.
All this time, I through 'intensity'  is tied to spawning monster's strength(Because I've  been made a kind like this mod before, and it worked well. ).

**Comment by JettLand** [comment]
@Regalis11 Hey,dude.I've been tried your way.
It works,but...there still have some problems,that makes me even more confused...
When if I set the 'chooserandom=true',and there is more than one event in the set,a strange phenomenon will occur:**Only the bottom line event will take effect.**
At the same time,some eventsets didn't work which one's 'chooserandom=false'(I tested many times to confirm that).
I tried to insert a 'commonness' function into the eventset,syntax correct,but it didn't work.
And then,I tried to nest an evenset in a evenset,and set 'chooserandom=true',still only bottom line event takes effect.
Is the algorithm of the event controller wrong?

**This is the file I reedited:**
[EnhanceRandomevents.txt](https://github.com/Regalis11/Barotrauma/files/4298279/EnhanceRandomevents.txt)
